arch/arm64/boot/dts/rockchip/overlays/rk3399-spi1-cs1-slb9670.dts

```diff
@@ -0,0 +1,40 @@
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/pinctrl/rockchip.h>
+
+/ {
+    metadata {
+        title = "Enable Infineon SLB9670 TPM on SPI1";
+        category = "misc";
+        compatible = "rockchip,rk3399";
+        exclusive = "GPIO1_B2", "GPIO4_D5";
```

Comment:
Those pins are now configured as SPI, so overlays trying to use them as GPIO or I2C should fail.

Adding extra checks to detect about situation and also allowing multiple SPI devices to share a bus is very complex. We are OK with more false positives here, since any complex configuration can be accomplished with a custom overlay, and at that point it is the user's responsibility to make sure their configuration is valid.